### PR TITLE
Fix configuration for persistence-query test deps (fix #22342)

### LINKED ISSUE #22342: Persistence-query depends on testkit JARs not in test scope
see http://search.maven.org/#artifactdetails|com.typesafe.akka|akka-persistence-query-experimental_2.11|2.4.17|jar

The dependencies look like this:

```
        <dependency>
            <groupId>com.typesafe.akka</groupId>
            <artifactId>akka-testkit_2.11</artifactId>
            <version>2.4.17</version>
        </dependency>
        <dependency>
            <groupId>com.typesafe.akka</groupId>
            <artifactId>akka-stream-testkit_2.11</artifactId>
            <version>2.4.17</version>
        </dependency>
```

But these JARs should be in test scope instead.

### PR DESCRIPTION
**Pull request by hseeberger**
Fixes #22342

### CODE REVIEW

File: project/AkkaBuild.scala [hunk 1]
```diff
@@ -216,6 +215,74 @@ object AkkaBuild extends Build {
...
+      id = s"akka-http-$name-experimental",
+      base = file(s"akka-http-marshallers-java/akka-http-$name"),
+      dependencies = Seq(http)
+    )
+
+  lazy val parsing = Project(
+    id = "akka-parsing",
+    base = file("akka-parsing")
+  )
+
```
Comment (patriknw) [inline project/AkkaBuild.scala]: what is this, why are http added here?

**Review by patriknw — CHANGES_REQUESTED** [review]

File: project/AkkaBuild.scala (hunk 1 above)
Comment (hseeberger) [inline project/AkkaBuild.scala]: Looks something went wrong when I tried to fix the merge conflicts ... thanks for noticing. I'll fix it again and update the PR.

**Comment by hseeberger** [comment]
@patriknw issue addressed

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
I'll backport

File: project/AkkaBuild.scala [hunk 2]
```diff
@@ -200,8 +200,7 @@ object AkkaBuild extends Build {
     dependencies = Seq(
       stream,
       persistence % "compile;provided->provided;test->test",
-      testkit % "compile;test->test",
-      streamTestkit % "compile;test->test")
+      streamTestkit % "test")
```
Comment (mr-git) [inline project/AkkaBuild.scala]: Just a curiosity: is there any benefit of using `% "test"` over `% Test`?

**Pull request merged**